pom.xml

```diff
@@ -16,25 +16,27 @@
     <description>API for document OCR</description>
 
     <properties>
-        <java.version>11</java.version>
+        <java.version>21</java.version>
 
         <included.tests>unit-test</included.tests>
 
         <apache-commons-lang.version>2.6</apache-commons-lang.version>
         <environment-reader.version>1.3.10</environment-reader.version>
-        <structured-logging.version>1.9.26</structured-logging.version>
+        <structured-logging.version>3.0.3</structured-logging.version>
+        <!-- Needed for structured-logging -->
+        <jakarta.servlet.version>6.0.0</jakarta.servlet.version>
+
         <tess4j.version>5.8.0</tess4j.version>
         <jib-maven-plugin.version>3.3.0</jib-maven-plugin.version>
 
         <!-- Spring General -->
-        <spring.boot.version>2.7.9</spring.boot.version>
+        <spring.boot.version>3.1.3</spring.boot.version>
 
         <!-- Maven -->
-        <maven-compiler-plugin.version>3.8.1</maven-compiler-plugin.version>
-        <maven-surefire-plugin.version>2.22.2</maven-surefire-plugin.version>
-        <sonar-maven-plugin.version>3.7.0.1746</sonar-maven-plugin.version>
+        <maven-compiler-plugin.version>3.9.0</maven-compiler-plugin.version>
+        <maven-surefire-plugin.version>3.2.5</maven-surefire-plugin.version>
+        <sonar-maven-plugin.version>3.9.1.2184</sonar-maven-plugin.version>
```

Comment:
These don't seem to be the latest versions, could we move to the latest versions?

Reply:
Actually `sonar-maven-plugin.version` seems to be defined in the parent pom so is this one redundant and could go?

Reply:
thanks @rafahop I have removed the overriden sonar maven plugin config.

Reply:
Thanks for removing the overridden property. Should we upgrade the other 2 plugins to the latest version? `maven-compiler-plugin` and `maven-surefire-plugin`

Reply:
thanks @rafahop I have updated them.